thewalrus/_hafnian.py

```diff
@@ -374,3 +378,38 @@ def hafnian_banded(A, loop=False, rtol=1e-05, atol=1e-08):
                 )
 
     return loop_haf[tuple(range(1, n + 1))]
+
+
+@jit(nopython=True)
+def one_det(B): # pragma: no cover
+    """ Calculates the determinant of an antisymmetric matrix with entries distributed
+    according to a normal distribution with scale equal to the entries of the symmetric matrix
+    given as input.
+
+    Args:
+        B (array): symmetric matrix
+
+    Returns:
+        (float): determinant of the samples antisymmetric matrix
+    """
+    mat = np.empty_like(B,dtype=np.float64)
+    n,m = B.shape
+    for i in range(n):
+        for j in range(m):
+            mat[i,j] = B[i,j]*np.random.normal()
+            mat[j,i] = -mat[i,j]
+    return np.linalg.det(mat)
+
+@jit(nopython=True)
+def hafnian_approx(A, num_samples = 1000): # pragma: no cover
+    """ Calculates Barvinok's approximation of the hafnian for a non-negative symmetric matrix.
```

Comment:
Should we copy over the old C++ docstring?
```suggestion
    """Returns the approximation to the hafnian of a matrix with non-negative entries.

    The approximation follows the stochastic Barvinok's approximation allowing the
    hafnian can be approximated as the sum of determinants of matrices.
    The accuracy of the approximation increases with increasing number of iterations. 
```